include/aws/crt/Optional.h

```diff
@@ -56,7 +56,8 @@ namespace Aws
             {
                 if (m_value)
                 {
-                    m_value->~T();
+                    *m_value = std::forward<U>(u);
```

Comment:
Doesn't have to be now, but we should have unit tests for this type, to catch stuff like this.